src/cockpit/389-console/src/lib/monitor/serverMonitor.jsx

```diff
@@ -1,201 +1,550 @@
+import {
+    faSyncAlt
+} from '@fortawesome/free-solid-svg-icons';
 
 export class ServerMonitor extends React.Component {
     constructor (props) {
         super(props);
+
+        let initChart = [];
+        for (let idx = 1; idx <= 10; idx++) {
+            initChart.push({name: '', x: idx.toString(), y: 0});
+        }
+
         this.state = {
             activeKey: 1,
+            count: 10,
+            port: 389,
+            secure_port: 636,
+            conn_highmark: 1000,
+            cpu_tick_values: [25, 50, 75, 100],
+            mem_tick_values: [25, 50, 75, 100],
+            conn_tick_values: [250, 500, 750, 1000],
+            mem_ratio: 0,
+            chart_refresh: "",
+            initChart: initChart,
+            cpuChart: [...initChart],
+            memVirtChart: [...initChart],
+            memResChart: [...initChart],
+            connChart: [...initChart],
         };
         this.handleNavSelect = this.handleNavSelect.bind(this);
+        this.startRefresh = this.startRefresh.bind(this);
+        this.refreshCharts = this.refreshCharts.bind(this);
+        this.stopRefresh = this.stopRefresh.bind(this);
     }
 
     componentDidMount() {
         this.props.enableTree();
+        this.refreshCharts();
+        this.startRefresh();
+    }
+
+    componentWillUnmount() {
+        this.stopRefresh();
     }
 
     handleNavSelect(key) {
         this.setState({ activeKey: key });
     }
 
+    getPorts() {
+        let cmd = [
+            "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
+            "config", "get", "nsslapd-port", "nsslapd-secureport"
+        ];
+        log_cmd("getPorts", "Get the server port numbers", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message" })
+                .done(content => {
+                    let config = JSON.parse(content);
+                    this.setState({
+                        port: config.attrs['nsslapd-port'][0],
+                        secure_port:  config.attrs['nsslapd-secureport'] === undefined
+                            ? "-1"
+                            : config.attrs['nsslapd-secureport'][0],
+                    });
+                });
+    }
+
+    resetChartData() {
+        this.setState({
+            conn_highmark: 1000,
+            cpu_tick_values: [25, 50, 75, 100],
+            mem_tick_values: [],
+            conn_tick_values: [250, 500, 750, 1000],
+            cpuChart: [...this.state.initChart],
+            memVirtChart: [...this.state.initChart],
+            memResChart: [...this.state.initChart],
+            connChart: [...this.state.initChart],
+        });
+    }
+
+    refreshCharts() {
+        let cmd = [
+            "ps", "-ef", "|", "grep", "slapd-" + this.props.serverId,
+            "|", "grep", "dirsrv"
+        ];
+        cmd = "ps -ef | grep slapd-" + this.props.serverId + " | grep dirsrv";
```

Comment:
Second assignment to the variable "cmd".
You might want to remove the first one.